model/common/src/icon4py/model/common/grid/icon.py

```diff
@@ -79,61 +79,57 @@ def __init__(
 @dataclasses.dataclass
 class GlobalGridParams:
     grid_shape: Final[GridShape | None] = None
-    _num_cells: int | None = None
-    _mean_cell_area: float | None = None
     radius: float = constants.EARTH_RADIUS
+    num_cells: int
+    mean_cell_area: float
 
-    @classmethod
-    def from_mean_cell_area(
-        cls,
-        mean_cell_area: float,
+    def __init__(
+        self,
+        *,
         grid_shape: GridShape | None = None,
-        num_cells: int | None = None,
         radius: float = constants.EARTH_RADIUS,
-    ):
-        return cls(
-            grid_shape,
-            num_cells,
-            mean_cell_area,
-            radius,
-        )
+        num_cells: int | None = None,
```

Comment:
Do this two parameters even make sense? What about these this test?
```
def test_num_cell_arg():
    shape = GridShape(geometry_type=ICOSAHEDRON), GridSubdivison(root = 4, level=9))
    params = GlobalGridParams(grid_shape=shape, num_cells = 2)
    assert 2 == params.num_cells
```

Reply:
Yeah, I suppose this particular combination isn't super realistic, but it just shows that it gives priority to the explicit `num_cells` parameter. The more meaningful changes are coming in https://github.com/C2SM/icon4py/pull/848, so may be worth waiting until that one for a longer discussion on what should be preferred/allowed and what not.